src/CommandProcessor.php

```diff
@@ -137,8 +141,19 @@ public function validateRunAndAlter(
             return $validated;
         }
 
-        // Run the command, alter the results, and then handle output and status
-        $result = $this->runCommandCallback($commandCallback, $commandData);
+        $replaceDispatcher = new ReplaceCommandHookDispatcher($this->hookManager(), $names);
+        if ($this->logger) {
```

Comment:
Minor style point: avoid unnecessary `if`s. It doesn't hurt to `setLogger()` with null, so there's no need to test here.